impl/README.md

```diff
@@ -0,0 +1,188 @@
+
+### Detailed version
+
+To discuss runtime API we are going to use a couple of workflow:
+- [listen.yaml](../examples/events/src/main/listen.yaml), which waits for an event reporting a temperature greater than 38
+- [emit.yaml](../examples/events/src/main/emit.yaml), which emits events with a certain temperature, specified as workflow parameter.
+
+Here is a summary of what we are trying to do: We will start the `listen.yaml` workflow, which will be completed when it receives an event with the proper temperature. It won't block the main thread while waiting for it. Then, we will send an event with a lower temperature that will be ignored. Finally, we will send an event with a higher temperature to complete the waiting workflow. 
+
+The first step is to create a [WorkflowApplication](core/src/main/java/io/serverlessworkflow/impl/WorkflowApplication.java) instance. An application is an abstraction that allows customization of different aspects of the workflow execution (for example, change the default `ExecutorService` for thread spawning)
+
+Since `WorkflowApplication` implements `Autocloseable`, we better use a try...finally block, ensuring any resource that the workflow might have used is freed when done. 
+
+`try (WorkflowApplication appl = WorkflowApplication.builder().build())`
+
+Once we have the application object, we use it to parse our definition examples. To load each workflow definition, we use the `readFromClasspath` helper method defined in [WorkflowReader](api/src/main/java/io/serverlessworkflow/api/WorkflowReader.java) class.
+
+```java
+      WorkflowDefinition listenDefinition =
+          appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("listen.yaml"));
+      WorkflowDefinition emitDefinition =
+          appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("emit.yaml")); 
+```
+
+A [WorkflowDefinition](core/src/main/java/io/serverlessworkflow/impl/WorkflowDefinition.java) object is immutable and, therefore, thread-safe. It is used to execute as many workflow instances as desired. 
+
+To execute a workflow, we first create a [WorkflowInstance](core/src/main/java/io/serverlessworkflow/impl/WorkflowInstance.java) object (its initial status is PENDING) and then invoke the `start` method on it (its status is changed to RUNNING). The `start` method returns a [CompletableFuture](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html), which we use to indicate that a log message should be printed when the workflow is completed.
+
+```java
+  WorkflowInstance waitingInstance = listenDefinition.instance(Map.of());
+      waitingInstance
+          .start()
+          .thenAccept(node -> logger.info("Waiting instance completed with result {}", node));
+```
+
+As soon as the workflow execution reach the point where it waits for events to arrive, control is returned to the calling thread. Therefore, we can execute another workflow instance while the first one is waiting. 
```

Comment:
The second phrase could start as "Since execution is non-blocking, ..." instead of "Therefore"